datalad/runner/tests/test_witless_runner.py

```diff
@@ -311,3 +311,34 @@ def test_too_long():
                 protocol=StdOutCapture
             )
         cml.assert_logged('.*use.*ulimit.*')
+
+
+def test_path_to_str_conversion():
+    # Regression test to ensure that Path-objects are converted into strings
+    # before they are put into the environment variable `$PWD`
+    runner = Runner()
+    adjusted_env = runner._get_adjusted_env(
+        cwd=Path("/a/b/c"),
+        env=dict(some_key="value")
+    )
+    assert "/a/b/c" == adjusted_env['PWD']
+
+
+@with_tempfile(mkdir=True)
+def test_environment(temp_dir_path=None):
```

Comment:
This test makes me think we have not covered the case of passing cwd to the constructor and (a different one) to the method. We probably should test that behavior, too.

Reply:
Both are tested. Line 336 -337 test passing to method, line 341-342 test passing to the constructor. Or did I missunderstand you?

Reply:
Yes, you did. We do not test what happens if the caller sets (possibly different) `cwd` via both - constructor *and* method. This is only testing the case where the caller does either one of those.

Reply:
In code: `Runner(cwd="one/path").run(cwd="another/path")` - what happens?

Reply:
The method provided variables take precedence over the constructor provided. Don't ask me what the original authors intended with this behavior, it might be suited to reusing "pre-configured" runners multiple times with minimal source code 

> One more thing, @christian-monch: Looks like the command quoting in the test (`py2cmd( ...`) is wrong for windows. We do have a helper for that somewhere (trying to find).

`py2cmd` is the helper. I copied it from an older version of `datalad.tests.test_witless_runner.py`, and just added the additional argumnets, which are not used in this context. It works on some windows environments, and not on others. I think the reason might be in the selection of string-delimiters within the python code-string. It seems that `Popen` on some windows environments uses `"` to delimit the arguments when they are passed down to the system. That conflicts with using them in the python code (the error message does not reflect the string that is passed down, this would only contain one double-quote around `PWD`, i.e. `["PWD"]`).

So I will leave `py2cmd` in its original state until we figure out what goes on.